src/Streamer.cpp

```diff
@@ -48,102 +58,129 @@ FileWriter::Streamer::connect(std::string TopicName) {
+FileWriter::Streamer::pollAndProcess(FileWriter::DemuxTopic &MessageProcessor) {
 
   try {
     // wait for connect() to finish
-    if (IsConnected.valid()) {
-      if (IsConnected.wait_for(std::chrono::milliseconds(100)) !=
+    if (RunStatus > StreamerStatus::IS_CONNECTED) {
+      // Do nothing
+    } else if (ConsumerCreated.valid()) {
+      if (ConsumerCreated.wait_for(std::chrono::milliseconds(100)) !=
           std::future_status::ready) {
-        LOG(Sev::Critical, "... still not ready");
-        return ProcessMessageResult::OK();
+        LOG(Sev::Warning,
+            "Not yet done setting up consumer. Defering consumption.");
+        return ProcessMessageResult::OK;
       }
-      RunStatus = IsConnected.get();
+      auto Temp = ConsumerCreated.get();
+      RunStatus = Temp.first;
+      Consumer = std::move(Temp.second);
+    } else {
+      throw std::runtime_error(
+          "Failed to set-up process for creating consumer.");
     }
+  } catch (std::runtime_error &Error) {
+    throw Error;
   } catch (std::exception &Error) {
-    LOG(Sev::Critical, "{}", Error.what());
+    LOG(Sev::Critical, "Got an exception when waiting for connection: {}",
+        Error.what());
+    throw Error;
   }
 
   // make sure that the connection is ok
   // attention: connect() handles exceptions
-  if (!RunStatus.connectionOK()) {
+  if (RunStatus < StreamerStatus::IS_CONNECTED) {
     throw std::runtime_error(Err2Str(RunStatus));
   }
 
-  // consume message and make sure that's ok
+  // Consume message and exit if we are beyond a message timeout
   KafkaW::PollStatus Poll = Consumer->poll();
   if (Poll.isEmpty() || Poll.isEOP()) {
-    if ((Options.StopTimestamp.count() > 0) &&
-        (systemTime() > (Options.StopTimestamp + Options.AfterStopTime))) {
-      LOG(Sev::Info, "Close topic {} after time expired",
-          MessageProcessor.topic());
+    if ((Options.StopTimestamp.count() > 0) and
+        (Options.ConsumerTimeout < (systemTime() - LastMessageTimestamp))) {
+      LOG(Sev::Info, "Stop stream timeout for topic \"{}\" reached. {} ms "
+                     "passed since last message received.",
+          MessageProcessor.topic(),
+          (systemTime() - LastMessageTimestamp).count());
       Sources.clear();
-      return ProcessMessageResult::STOP();
+      return ProcessMessageResult::STOP;
     }
-    return ProcessMessageResult::OK();
+    return ProcessMessageResult::OK;
   }
   if (Poll.isErr()) {
-    return ProcessMessageResult::ERR();
+    return ProcessMessageResult::ERR;
   }
 
-  // convert from KafkaW to Msg
-  Msg Message(Msg::fromKafkaW(Poll.isMsg()));
-  if (Message.type == MsgType::Invalid) {
-    return ProcessMessageResult::ERR();
+  // Convert from KafkaW to FlatbufferMessage, handles validation of flatbuffer
+  auto KafkaMessage = Poll.isMsg();
+  std::unique_ptr<FlatbufferMessage> Message;
+  try {
+    Message.reset(new FlatbufferMessage(
```

Comment:
I didn't really understand what you said in the code review at this point, declaring `Message` outside the `try` statement doesn't, and shouldn't, stop you from using `make_unique` here.

Reply:
I do not remember what I said but I remember what I meant: I can not put the declaration in the try statement as I need the Message after the try statement as well. I have changed to using `std::make_unique<>()` though.